wiki/ccxt.pro.manual.md

```diff
@@ -1198,6 +1198,72 @@ cancel_all_orders_ws(string $symbol, $params = array ())
 - this method is a work in progress now (may be unavailable)
 ```
 
+### Custom handler
+
+If you want to have an access to raw incoming messages and use your custom handlers, you can override exchange's `handleMessage/handle_message` method, like:
+
+<!-- tabs:start -->
+#### **Javascript**
+```javascript
+function myHandler(ws, data){
```

Comment:
I'm not sure this is the best way, I think you should be calling the parent method to keep things working as expected (like the returned values)

Reply:
right, i've amended them, is now ok?

Reply:
@ttodua no, did you test it? 😅 that will create an infinite recursion, I was suggesting something like this (but in js)

```
class my_exchange(ccxt.pro.phemex):
    def handle_message(self, client, message):
        print("got message", message) # this is the raw update
        super().handle_message(client, message)

exchange = my_exchange({...})
``

Reply:
I think using a proper inheritance approach is better here

Reply:
maybe i've messed up example during commits, initially i had lambda funcs in example and then in another commit changed them to external method. give me a moment plz

Reply:
ah damn, right, in debugger it goes normally in first cycle, but then it over and over recurses, i've not followed it yesterday, stupid mistake  :D

Reply:
about the above - inheritance is generally a better approach, but we should offer/show both approaches, because overriding exchange is no longer `ccxt.xyzExchange` and applications might be instantiating ccxt exchanges directly, and inheritance is not fit for all use-cases.

i've pushed an update again, maybe now it's ok?